### PR TITLE
[13.0][ADD] survey_legal: New module 13.0

### PR DESCRIPTION
**Pull request by chienandalu**
This module allows you to include a legal terms acceptance checkbox in your surveys. You
have the ability to customize this feature for each individual survey.

cc @Tecnativa TT40980

please review @pedrobaeza @victoralmau

### CODE REVIEW

**Review by victoralmau — APPROVED** [review]
Code and functionality review OK.

**Review by pedrobaeza — COMMENTED** [review]
Thanks for this, David. I miss the same kind of "footprint" that we have on CRM that the user has accepted the legal terms on the answers. Can you please add it?

**Comment by chienandalu** [comment]
Added metadata recording. Only tests would be missing

File: survey_legal/views/survey_user_input_views.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<odoo>
+    <record id="survey_user_input_view_form" model="ir.ui.view">
+        <field name="model">survey.user_input</field>
+        <field name="inherit_id" ref="survey.survey_user_input_view_form" />
+        <field name="arch" type="xml">
+            <field name="user_input_line_ids" position="after">
+                <field name="survey_legal_terms" invisible="1" />
+                <group
+                    name="metadata"
+                    attrs="{'invisible': ['|', ('survey_legal_terms', '!=', True), ('state', '=', 'new')]}"
```
Comment (pedrobaeza) [inline survey_legal/views/survey_user_input_views.xml]: Isn't better to simply show this if filled?

Comment (chienandalu) [reply survey_legal/views/survey_user_input_views.xml]: Yes, it would TBH :sweat_smile:

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**